### PR TITLE
 firefox, firefox-bin: 88.0 -> 88.0.1, firefox-esr: 78.10.0esr -> 78.10.1esr

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

88.0.1:
Release notes are not updated yet.

https://www.mozilla.org/en-US/firefox/88.0.1/releasenotes/

78.10.1esr:
- Moderate security fix (Windows only)
- Other fix (Windows only)

https://www.mozilla.org/en-US/firefox/78.10.1/releasenotes/
https://www.mozilla.org/en-US/security/advisories/mfsa2021-18/

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/browsers/firefox/packages.nix [hunk 1]
```diff
@@ -7,10 +7,10 @@ in
 rec {
   firefox = common rec {
     pname = "firefox";
-    ffversion = "88.0";
+    ffversion = "88.0.1";
     src = fetchurl {
       url = "mirror://mozilla/firefox/releases/${ffversion}/source/firefox-${ffversion}.source.tar.xz";
-      sha512 = "f58f44f2f0d0f54eae5ab4fa439205feb8b9209b1bf2ea2ae0c9691e9e583bae2cbd4033edb5bdf4e37eda5b95fca688499bed000fe26ced8ff4bbc49347ce31";
+      sha512 = "0sakkl4bsjcl7bq08dgvmqjpcz2i5h7bmbf4miv61x25zg50g3k5rppcmw5p7f4hl5z9dx36dng3z0qcffpj7gfhdf257x41fazrmz2";
```
Comment (mweinelt) [inline pkgs/applications/networking/browsers/firefox/packages.nix:13]: Thanks!

Comment (taku0) [reply pkgs/applications/networking/browsers/firefox/packages.nix:13]: It is generated by `update.nix`, which relies on generic `update-source-version` to guess the hash. We should update `update.nix` to use `SHA512SUMS`, perhaps in separate PR.

Comment (taku0) [reply pkgs/applications/networking/browsers/firefox/packages.nix:13]: Created #121761.

Comment (taku0) [reply pkgs/applications/networking/browsers/firefox/packages.nix]: Updated the hash.

**Pull request merged**